internal/datastore/crdb/crdb.go

```diff
@@ -90,20 +90,20 @@ func newCRDBDatastore(url string, options ...Option) (datastore.Datastore, error
 	}
 	config.writePoolOpts.ConfigurePgx(writePoolConfig)
 
-	initCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	initCtx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
```

Comment:
why the change? due to the connect rate making the initial population slower?

This could mean that with a big enough pool, the pod may fail the kube health checks and get killed before it even finishes populating the pool?

I understand the need for the rate limiter but it would seem like now folks have to make sure their deployments have readiness probes set in proportion to how long will it take for the pool to populate, which depends on the configuration arguments. The operator would have to be clever about this moving forward

Reply:
Yes that's exactly right. 

The good news is that you'll get the maximum acquisition rate on initial pod start, so if your timeout isn't long enough you should know right away. We may need to make this configurable in the future, but for now this seemed to be enough even at a high connection count / low connection rate.

The default is 1 connection / second / pool, so in 5 minutes the pod can make 600 connections to the database.